.github/workflows/ci.yml

```diff
@@ -272,6 +282,37 @@ jobs:
             ./build_tools/cmake/ctest_all.sh \
             "${BUILD_DIR}"
 
+  test_all_arm64:
+    needs: [setup, build_all_arm64]
```

Comment:
As `test_all_arm64` is the only job that follows `build_all_arm64`, we could merge the jobs, like how `build_test_all_macos_arm64` is set up. Splitting the jobs is useful when multiple following jobs want to use the build artifacts, but otherwise it just adds overhead (upload/download, wait for runner, etc.).

Reply:
I can do that. One thought though. If we'd like to extend with further arm64 server jobs in the future, isn't it convenient to keep them separated? (One example that has been discussed are the benchmark jobs.)

Reply:
The benchmark jobs do need the `iree-benchmark-module` tool compiled for the target platform, which is handled by the `build_benchmark_tools` job. That has a matrix here: https://github.com/openxla/iree/blob/668c020cc04d66a60b31a6cda46ebdf3b854e952/.github/workflows/build_benchmark_tools.yml#L61-L79

Since that just builds runtime tools, it takes ~3 minutes.

Reply:
Ok, so if I understood correctly the iree-compiler built as part of the `build_all` job is used to cross-compile everything that is going to be benchmark:ed?

Adding another target under [iree/.github/workflows/build_benchmark_tools.yml](https://github.com/openxla/iree/blob/668c020cc04d66a60b31a6cda46ebdf3b854e952/.github/workflows/build_benchmark_tools.yml#L61-L79) would create the executable needed to run that benchmark... About right?

Reply:
Yep! @pzread is more familiar with `build_e2e_test_artifacts` than me though.

Reply:
Yes, that's right

Reply:
Makes a lot of sense. Will do.

Reply:
Done. I removed the changes in `build_all.yml` since they're not needed for this change.

Reply:
Seems to pass. Let's see how what the build time drops to once the ccache is written.